### ISSUE TITLE
cryptic error about constants when there are no constants

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1442]
Code:

``` javascript
function f(array) {
  array.forEach((item) => console.log(item, this))
}
```

Expected result: Maybe a warning about an unknown this type?

Actual result: 

```
input0:1: WARNING - All constants must be typed. The compiler could not infer the type of this constant. Please use an explicit type annotation.
function f(array) {
                  ^
```

I guess it's warning about the generated code from the fat arrow function?

http://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(array)%2520%257B%250A%2520%2520array.forEach((item)%2520%253D%253E%2520console.log(item%252C%2520this))%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_ES5_STRICT%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26DISAMBIGUATE_PROPERTIES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26REWRITE_FUNCTION_EXPRESSIONS%3D1%26TRANSPILE%3D1

**Comment by MatrixFrog** [comment]
I believe ef5a76bedf261753729887b3784d17cbd16aa94c should fix this one as well.
And yes, we translate =>s to something like

`/** @const */ var $jscomp$this = this;`
`function() {`
`  console.log(item, this);`
`}`

so if you have an unknown "this" you'll get a warning. As far as I've seen the new conformance-check version of this check is a lot better / less noisy.

**Comment by MatrixFrog** [comment]
Sorry, nevermind. There is a similar (off by default) warning in the old type inference but I'm still seeing the NTI one.
